File: repos_estudiantes_TARQUI_segunda_entrega/QUISPE_MALDONADO_CINTHIA_JAZMIN/COMPOSICION-AGREGACION/EJERCICIO-1/Ejercicio1.py
class Habitacion:
    def __init__(self,nombre, tamano):
        self.nombre=nombre
        self.tamano=tamano
    def get_nombre(self):
        return self.nombre
    def get_tamano(self):
        return self.tamano
    def set_nombre(self, nombre):
        self.nombre=nombre
    def set_tamano(self, tamano):
        self.tamano=tamano
    def mostrar_info(self):
        print(f" Nombre: {self.nombre}")
        print(f" Tamano: {self.tamano} m²")
class Casa:
    def __init__(self,direccion):
        self.direccion=direccion
        self.habitaciones=[]
    def get_direccion(self):
        return self.direccion
    def set_direccion(self, direccion):
        self.direccion=direccion
    def agregar_habitacion(self, habitacion):
        self.habitaciones.append(habitacion)
    def mostrar_casa(self):
        print(f"Dirección de la casa: {self.direccion}")
        print("Habitaciones")
        for i in self.habitaciones:
            i.mostrar_info()

if __name__=="__main__":
    h1=Habitacion("Comedor",13)
    h2=Habitacion("Dormitorio",10)
    h3=Habitacion("Sala",15)
    h4=Habitacion("Cochera",20)
    h5=Habitacion("Cuarto de huespedes",9)
    h6=Habitacion("Cocina",12)
    h7=Habitacion("Bano",6)
    c1=Casa("Buenos Aires 154")
    c1.agregar_habitacion(h1)
    c1.agregar_habitacion(h2)
    c1.agregar_habitacion(h3)
    c1.agregar_habitacion(h4)
    c1.agregar_habitacion(h5)
    c1.agregar_habitacion(h6)
    c1.agregar_habitacion(h7)
    c1.mostrar_casa()

File: repos_estudiantes_ALVAREZ_tercera_entrega/ALBA_PLATA_BRYAN_ALBERTO/PracticaDOS/ejercicioCom1/Ejercicio1.java
import java.util.*;
class Habitacion {
    private String nombre;
    private double tamano;

    public Habitacion(String nombre, double tamano) {
        this.nombre = nombre;
        this.tamano = tamano;
    }

    public void mostrar_info() {
        System.out.println("Habitación: " + nombre + ", Tamaño: " + tamano + " m²");
    }
}

class Casa {
    private String direccion;
    private List<Habitacion> habitaciones;

    public Casa(String direccion) {
        this.direccion = direccion;
        habitaciones = new ArrayList<>();
    }

    public void agregar_habitacion(Habitacion h) {
        habitaciones.add(h);
    }

    public void mostrar_casa() {
        System.out.println("Dirección: " + direccion);
        for (Habitacion h : habitaciones) {
            h.mostrar_info();
        }
    }
}

public class Ejercicio1 {
    public static void main(String[] args) {
        Casa casa = new Casa("Av. 6 de Agosto #123");
        casa.agregar_habitacion(new Habitacion("Sala", 25));
        casa.agregar_habitacion(new Habitacion("Cocina", 12));
        casa.agregar_habitacion(new Habitacion("Dormitorio", 18));
        casa.mostrar_casa();
    }
    
}

